modules/apps/redirect/redirect-api/src/main/java/com/liferay/redirect/model/RedirectPatternUserAgent.java

```diff
@@ -0,0 +1,40 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.redirect.model;
+
+import java.util.regex.Pattern;
+
+/**
+ * @author Alicia García
+ */
+public class RedirectPatternUserAgent {
```

Comment:
I'd rename this to `RedirectPatternEntry`, and include an additional field for the replacement.